app/features/ChainDataSlice.ts

```diff
@@ -1,11 +1,62 @@
-import { BlockSummary, ConsensusStatus } from '@concordium/node-sdk';
+import { ConsensusStatus, KeysMatching } from '@concordium/node-sdk';
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { Dispatch } from 'redux';
-import { getBlockSummary, getConsensusStatus } from '~/node/nodeRequests';
+import { getConsensusStatus } from '~/node/nodeRequests';
+import type { RootState } from '~/store/store';
+import { pipe } from '~/utils/basicHelpers';
+import { orUndefined } from '~/utils/functionHelpers';
+
+type CSDateKey = KeysMatching<ConsensusStatus, Date>;
+type CSBigIntKey = KeysMatching<ConsensusStatus, bigint>;
+
+type SerializableConsensusStatus = Omit<
+    ConsensusStatus,
+    keyof CSDateKey | keyof CSBigIntKey
+> &
+    { [P in keyof CSDateKey]: number } &
+    { [P in keyof CSBigIntKey]: string };
+
+function toSerializableCS(cs: ConsensusStatus): SerializableConsensusStatus {
+    return Object.entries(cs).reduce((acc, [key, value]) => {
+        if (typeof value === 'bigint') {
+            return { ...acc, [key]: value.toString() };
+        }
+        if (value instanceof Date) {
+            return { ...acc, [key]: value.getTime() };
+        }
+        return { ...acc, [key]: value };
+    }, {} as SerializableConsensusStatus);
+}
+
+const csDates: CSDateKey[] = ['genesisTime', 'currentEraGenesisTime'];
+const csBigInts: CSBigIntKey[] = [
+    'epochDuration',
+    'slotDuration',
+    'bestBlockHeight',
+    'lastFinalizedBlockHeight',
+    'finalizationCount',
+    'blocksVerifiedCount',
+    'blocksReceivedCount',
+    'protocolVersion',
+];
+
+function toOriginalCS(scs: SerializableConsensusStatus): ConsensusStatus {
```

Comment:
Is there a reason you didn't just use stringify/parse for this :thinking:

Reply:
We could stringify, but we need to parse manually either way? Or am I missing something? How would you otherwise identify bigints/dates to be converted?